wrangler-transform/src/main/java/io/cdap/wrangler/Wrangler.java

```diff
@@ -511,14 +471,89 @@ private Object getValue(StructuredRecord input, String fieldName) {
+    if (inputSchema != null) {
+      // Check the existence of field in input schema
+      Schema.Field inputSchemaField = inputSchema.getField(config.field);
+      if (inputSchemaField == null) {
+        collector.addFailure(String.format("Field '%s' must be present in input schema.", config.field), null)
+          .withConfigProperty(Config.NAME_FIELD);
+      }
+    }
+  }
+
+  private RecipeParser initializeAndGetParser(
+    StageContext context) throws DirectiveLoadException, DirectiveParseException {
+    registry = new CompositeDirectiveRegistry(new SystemDirectiveRegistry(), new UserDirectiveRegistry(context));
+    registry.reload(context.getNamespace());
+
+    String directives = config.directives;
+    if (config.udds != null && !config.udds.trim().isEmpty()) {
+      directives = String.format("#pragma load-directives %s;%s", config.udds, config.directives);
+    }
+
+    RecipeParser recipe = new GrammarBasedParser(context.getNamespace(), new MigrateToV2(directives).migrate(),
+                                                 registry);
+
+    // Make a call to retrieve the directive config and create a context.
+    // In this plugin, a call is made to the service at runtime.
+    // NOTE: This has to be moved to pre-start-plugin phase so that the information
+    // it's not heavily parallelized to crash the service.
+    try {
+      URL url = getDPServiceURL(context, CONFIG_METHOD);
+      if (url != null) {
+        ConfigDirectiveContext dContext = new ConfigDirectiveContext(url);
+        recipe.initialize(dContext);
+      } else {
+        // this is normal in cloud environments
+        LOG.info(String.format("Stage:%s - The Dataprep service is not accessible in this environment. "
+                                 + "No aliasing and restriction will be applied.", getContext().getStageName()));
+        recipe.initialize(null);
+      }
+    } catch (IOException | URISyntaxException e) {
+      // If there is a issue, we need to fail the pipeline that has the plugin.
+      throw new IllegalArgumentException(
+        String.format("Stage:%s - Issue in retrieving the configuration from the service. %s",
+                      getContext().getStageName(), e.getMessage()), e
+      );
+    }
+    return recipe;
+  }
+
   /**
    * Retrieves the base url from the context and appends method to value to the final url.
    *
    * @param method to be invoked.
    * @return fully formed url to the method.
    */
-  private URL getDPServiceURL(String method) throws URISyntaxException, MalformedURLException {
-    URL url = getContext().getServiceURL(Contexts.SYSTEM, APPLICATION_NAME, SERVICE_NAME);
+  private URL getDPServiceURL(ServiceDiscoverer discoverer,
+                              String method) throws URISyntaxException, MalformedURLException {
+    URL url = discoverer.getServiceURL(Contexts.SYSTEM, APPLICATION_NAME, SERVICE_NAME);
```

Comment:
what if discoverer is null ?

Reply:
The discoverer here is the StageContext, it cannot be null